railties/lib/rails/command/actions.rb

```diff
@@ -10,23 +10,21 @@ def set_application_directory!
         Dir.chdir(File.expand_path("../..", APP_PATH)) unless File.exist?(File.expand_path("config.ru"))
       end
 
-      def require_application_and_environment!
-        require_application!
-        require_environment!
-      end
-
       def require_application!
         require ENGINE_PATH if defined?(ENGINE_PATH)
+        require APP_PATH if defined?(APP_PATH)
+      end
 
-        if defined?(APP_PATH)
-          require APP_PATH
-        end
+      def boot_application!
+        require_application!
+        Rails.application.require_environment! if defined?(APP_PATH)
       end
 
-      def require_environment!
-        if defined?(APP_PATH)
-          Rails.application.require_environment!
-        end
+      def load_environment_config!
+        require_application!
+        # Only run initializers that are in the :all group, which includes the
+        # :load_environment_config initializer.
+        Rails.application.initialize!(:_) if defined?(APP_PATH)
```

Comment:
Should this say `initialize!(:all)`? I don't see any special treatment for `:_` [here](https://github.com/jonathanhefner/rails/blob/20b541524463a1e4ed9e65a18b2c454affb3bbd4/railties/lib/rails/initializable.rb#L28) which is what I thought you were doing.

Reply:
I felt like `Rails.application.initialize!(:all)` would read as if we actually were running "all" initializers.  I chose `:_` to be a "non-value" (i.e. non-existent group) and to avoid clashing with any initializer groups from gems or the app.

Reply:
Ohhhh I follow now.